library/Zend/Form/Annotation/ElementAnnotationsListener.php

```diff
@@ -163,6 +163,11 @@ public function handleComposedObjectAnnotation($e)
                 $specification['type'] = 'Zend\Form\Fieldset';
             }
 
+            if (isset($elementSpec['spec']['options'])) {
+                $specification['options'] = isset($specification['options']) ? $specification['options'] : array();
+                $specification['options'] = array_merge($elementSpec['spec']['options'], $specification['options']);
```

Comment:
Actually, @malukenho, it's correct as specified; we're trying to set the `$specification['options']` and/or update it; if the conditional is changed as you specify, `$specification['options']` will be empty later.